interfaces/devdoc/threadpool_requirements.md

```diff
@@ -31,6 +31,12 @@ MOCKABLE_FUNCTION(, THANDLE(THREADPOOL), threadpool_create, EXECUTION_ENGINE_HAN
 MOCKABLE_FUNCTION(, int, threadpool_open, THANDLE(THREADPOOL), threadpool);
 MOCKABLE_FUNCTION(, void, threadpool_close, THANDLE(THREADPOOL), threadpool);
 
+MOCKABLE_FUNCTION(, THREADPOOL_WORK_ITEM_HANDLE, threadpool_create_work_item, THANDLE(THREADPOOL), threadpool, THREADPOOL_WORK_FUNCTION, work_function, void*, work_function_context);
+
+MOCKABLE_FUNCTION(, int, threadpool_schedule_work_item, THANDLE(THREADPOOL), threadpool, THREADPOOL_WORK_ITEM_HANDLE, work_item_context);
```

Comment:
>work_item_context [](http://example.com/codeflow?start=118&length=17)

poetry: Can remove the word "_context" here and in the next function. We typically use "context" along with arbitrary data that is passed along with callbacks (like `work_function_context` above), but not in a case like this (see the other functions here which take a `timer` or a `threadpool` and not a `timer_context`/`threadpool_context`.

Reply:
It's work_item_context in the original threadpool_schedule_work that we are now passing around. Earlier we used to create it and free inside the callback function. But now we are saving it by returning it to the caller from the create function

Reply:
Matt means that conceptually what gets passed here from the point of view of the user of this API is a work item handle. We generally use context for a pointer associated with something.

Looking from the point of view of a user of the APIs. For example for threadpool_schedule_work, user passes a work function and an opaque context associated with that work_function.

In the case of this API (threadpool_schedule_work_item), the user passes a work item handle, not a context associated with a work item handle.

So it should be named as such: work_itm or work_item_handle.